test/integration/causal-consistency/causal_consistency.prose.test.js

```diff
@@ -0,0 +1,259 @@
+'use strict';
+
+const { LEGACY_HELLO_COMMAND } = require('../../../src/constants');
+
+const { setupDatabase } = require('../shared');
+const { expect } = require('chai');
+
+const ignoredCommands = [LEGACY_HELLO_COMMAND, 'endSessions'];
+const test = { commands: { started: [], succeeded: [] } };
+
+// TODO(NODE-3882) - properly implement all prose tests and add missing cases 1, 8, 9, 11, 12
+describe('Causal Consistency - prose tests', function () {
+  before(function () {
+    return setupDatabase(this.configuration);
+  });
+
+  beforeEach(function () {
+    test.commands = { started: [], succeeded: [] };
+    test.client = this.configuration.newClient({ w: 1 }, { maxPoolSize: 1, monitorCommands: true });
+    test.client.on('commandStarted', event => {
+      if (ignoredCommands.indexOf(event.commandName) === -1) test.commands.started.push(event);
+    });
+
+    test.client.on('commandSucceeded', event => {
+      if (ignoredCommands.indexOf(event.commandName) === -1) test.commands.succeeded.push(event);
+    });
+
+    return test.client.connect();
+  });
+
+  afterEach(() => {
+    return test.client.close();
+  });
+
+  it(
+    '2. The first read in a causally consistent session must not send afterClusterTime to the server',
+    /**
+     * session = client.startSession(causalConsistency = true)
+     * document = collection.anyReadOperation(session, ...)
+     * capture the command sent to the server (using APM or other mechanism)
+     * assert that the command does not have an afterClusterTime
+     */
+    {
+      metadata: {
+        requires: { topology: ['replicaset', 'sharded'] },
+        // Skipping session leak tests b/c these are explicit sessions
```

Comment:
This is interesting, I would think this test still clean up the sessions it creates